### DISCUSSION TITLE
Getting 503 (Server Unavailable) error from CSLA

### DISCUSSION THREAD

**Discussion by ghost** [MarimerLLC/csla#2147] [Questions]
I am using CSLA 4.3.13.0 within a large complicated application. I have a CSLA WCF service with a number of different clients that connect to it.

One of those clients is a windows service running on the same server that is hosting the WCF service.  The windows service retrieves a list of CSLA objects, processes them one at a time and then saves the results one at a time by calling the .Save() method.

This arrangement has worked for many years on a number of different servers and on multiple instances of the application. However on one new server that we just upgraded to I am getting an error that I can't explain.

The CSLA client is able to retrieve the data that it needs, including multiple requests through a few different CSLA objects. However, when it tries to save the results it says that it gets an HTTP 503 error (meaning Service Unavailable). Once the error occurs the client will no longer access the server, even if previously successful calls are repeated. To get it to work at all the client needs to be restarted.

However, despite the 503 error, the service never actually goes down and other clients are able to access it simultaneously with no problems. Also, no 503 error appears in the IIS logs.

It feels like the server URL is somehow being lost or altered during execution, but I can't figure out how to check the actual address being used at runtime.

Please help.
Jacob

**Comment by rockfordlhotka** [comment]
Your best bet is probably to use something like fiddler to watch the traffic and see if you can identify the issue.

**Reply by ghost** [reply]
Thank you. I tried to use Fiddler to capture the traffic. However, I have bene unable to figure out how to get my service to use the Fiddler proxy. Do you have any suggestions?

--Jacob

**Reply by ghost** [reply]
I should clarify that I know how to configure proxy settings using more typical tools like an HTTPRequest. However I don't know how to set them for CSLA.

--Jacob

**Reply by rockfordlhotka** [reply]
It has been some years since I've used WCF.

My recollection is that WcfProxy (in CSLA) has two ways to configure.

1. Via your config file, because WcfProxy lets WCF configure itself based on the file
2. By subclassing WcfProxy and overriding the methods that create and configure the underlying WCF proxy

The _Using CSLA 4_ book series includes a book on configuring the data portal, which covers configuring WcfProxy in detail.
